crates/vault-registry/src/lib.rs

```diff
@@ -1190,41 +1284,38 @@ impl<T: Config> Pallet<T> {
         Ok(to_slash)
     }
 
-    pub(crate) fn increase_total_backing_collateral(amount: &Amount<T>) -> DispatchResult {
-        let new = TotalUserVaultCollateral::<T>::get()
-            .checked_add(&amount.amount())
-            .ok_or(Error::<T>::ArithmeticOverflow)?;
+    pub fn try_increase_total_backing_collateral(amount: &Amount<T>) -> DispatchResult {
+        let new = Self::get_total_user_vault_collateral(amount.currency())?.checked_add(&amount)?;
 
-        TotalUserVaultCollateral::<T>::set(new);
+        let limit = Self::get_collateral_ceiling(amount.currency())?;
+        ensure!(new.le(&limit)?, Error::<T>::CollateralCurrencyCeilingExceeded);
+
+        TotalUserVaultCollateral::<T>::insert(&amount.currency(), new.amount());
 
         Ok(())
     }
 
-    pub(crate) fn decrease_total_backing_collateral(amount: &Amount<T>) -> DispatchResult {
-        let new = TotalUserVaultCollateral::<T>::get()
-            .checked_sub(&amount.amount())
-            .ok_or(Error::<T>::ArithmeticUnderflow)?;
+    pub fn decrease_total_backing_collateral(amount: &Amount<T>) -> DispatchResult {
+        let new = Self::get_total_user_vault_collateral(amount.currency())?.checked_sub(amount)?;
 
-        TotalUserVaultCollateral::<T>::set(new);
+        TotalUserVaultCollateral::<T>::insert(&amount.currency(), new.amount());
 
         Ok(())
     }
 
-    /// returns the total locked collateral, _
+    /// returns the total locked collateral. Unused for now, but might want to expose that as rpc
     pub fn get_total_backing_collateral(
```

Comment:
Let's remove any unused code, if we do need it in the future it should be easy enough to add (can check git history) - this file is already quite big...